File: qrcodereaderview/src/test/java/com/dlazaro66/qrcodereaderview/QRToViewPointTransformerShould.java
package com.dlazaro66.qrcodereaderview;

import android.graphics.Point;
import android.graphics.PointF;

import com.google.zxing.ResultPoint;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class QRToViewPointTransformerShould {

  private QRToViewPointTransformer qrToViewPointTransformer;

  @Before
  public void setUp() throws Exception {
    qrToViewPointTransformer = new QRToViewPointTransformer();
  }

  @After
  public void tearDown() throws Exception {

  }

  @Test
  public void transformPortraitNotMirrorQRPointToViewPoint() throws Exception {
    ResultPoint qrPoint = new ResultPoint(100, 50);
    boolean isMirrorPreview = false;
    Orientation orientation = Orientation.PORTRAIT;
    Point viewSize = new Point(100, 200);
    Point cameraPreviewSize = new Point(200, 100);

    PointF result = qrToViewPointTransformer.transform(qrPoint, isMirrorPreview, orientation,
        viewSize, cameraPreviewSize);

    assertEquals(result.x, 50, 0.0f);
    assertEquals(result.y, 100, 0.0f);
  }

  @Test
  public void transformPortraitMirrorQRPointToViewPoint() throws Exception {
    ResultPoint qrPoint = new ResultPoint(0, 0);
    boolean isMirrorPreview = true;
    Orientation orientation = Orientation.PORTRAIT;
    Point viewSize = new Point(100, 200);
    Point cameraPreviewSize = new Point(200, 100);

    PointF result = qrToViewPointTransformer.transform(qrPoint, isMirrorPreview, orientation,
        viewSize, cameraPreviewSize);

    assertEquals(result.x, 100, 0.0f);
    assertEquals(result.y, 200, 0.0f);
  }

  @Test
  public void transformLandscapeNotMirrorQRPointToViewPoint() throws Exception {
    ResultPoint qrPoint = new ResultPoint(100, 50);
    boolean isMirrorPreview = false;
    Orientation orientation = Orientation.LANDSCAPE;
    Point viewSize = new Point(200, 100);
    Point cameraPreviewSize = new Point(200, 100);

    PointF result = qrToViewPointTransformer.transform(qrPoint, isMirrorPreview, orientation,
        viewSize, cameraPreviewSize);

    assertEquals(result.x, 100, 0.0f);
    assertEquals(result.y, 50, 0.0f);
  }

  @Test
  public void transformLandscapeMirrorQRPointToViewPoint() throws Exception {
    ResultPoint qrPoint = new ResultPoint(0, 0);
    boolean isMirrorPreview = true;
    Orientation orientation = Orientation.LANDSCAPE;
    Point viewSize = new Point(200, 100);
    Point cameraPreviewSize = new Point(200, 100);

    PointF result = qrToViewPointTransformer.transform(qrPoint, isMirrorPreview, orientation,
        viewSize, cameraPreviewSize);

    assertEquals(result.x, 0, 0.0f);
    assertEquals(result.y, 100, 0.0f);
  }
}

File: qrcodereaderview/src/test/java/com/dlazaro66/qrcodereaderview/QRToViewPointTransformerShould.kt
package com.dlazaro66.qrcodereaderview

import android.graphics.Point
import com.google.zxing.ResultPoint
import org.junit.After
import org.junit.Assert
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class QRToViewPointTransformerShould {
    private var qrToViewPointTransformer: QRToViewPointTransformer? = null
    @Before
    @Throws(Exception::class)
    fun setUp() {
        qrToViewPointTransformer = QRToViewPointTransformer()
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
    }

    @Test
    @Throws(Exception::class)
    fun transformPortraitNotMirrorQRPointToViewPoint() {
        val qrPoint = ResultPoint(100f, 50f)
        val isMirrorPreview = false
        val orientation = Orientation.PORTRAIT
        val viewSize = Point(100, 200)
        val cameraPreviewSize = Point(200, 100)
        val result = qrToViewPointTransformer!!.transform(qrPoint, isMirrorPreview, orientation,
                viewSize, cameraPreviewSize)!!
        Assert.assertEquals(result.x, 50f, 0.0f)
        Assert.assertEquals(result.y, 100f, 0.0f)
    }

    @Test
    @Throws(Exception::class)
    fun transformPortraitMirrorQRPointToViewPoint() {
        val qrPoint = ResultPoint(0f, 0f)
        val isMirrorPreview = true
        val orientation = Orientation.PORTRAIT
        val viewSize = Point(100, 200)
        val cameraPreviewSize = Point(200, 100)
        val result = qrToViewPointTransformer!!.transform(qrPoint, isMirrorPreview, orientation,
                viewSize, cameraPreviewSize)!!
        Assert.assertEquals(result.x, 100f, 0.0f)
        Assert.assertEquals(result.y, 200f, 0.0f)
    }

    @Test
    @Throws(Exception::class)
    fun transformLandscapeNotMirrorQRPointToViewPoint() {
        val qrPoint = ResultPoint(100f, 50f)
        val isMirrorPreview = false
        val orientation = Orientation.LANDSCAPE
        val viewSize = Point(200, 100)
        val cameraPreviewSize = Point(200, 100)
        val result = qrToViewPointTransformer!!.transform(qrPoint, isMirrorPreview, orientation,
                viewSize, cameraPreviewSize)!!
        Assert.assertEquals(result.x, 100f, 0.0f)
        Assert.assertEquals(result.y, 50f, 0.0f)
    }

    @Test
    @Throws(Exception::class)
    fun transformLandscapeMirrorQRPointToViewPoint() {
        val qrPoint = ResultPoint(0f, 0f)
        val isMirrorPreview = true
        val orientation = Orientation.LANDSCAPE
        val viewSize = Point(200, 100)
        val cameraPreviewSize = Point(200, 100)
        val result = qrToViewPointTransformer!!.transform(qrPoint, isMirrorPreview, orientation,
                viewSize, cameraPreviewSize)!!
        Assert.assertEquals(result.x, 0f, 0.0f)
        Assert.assertEquals(result.y, 100f, 0.0f)
    }
}